frontend/packages/dev-console/src/components/pipelines/pipeline-overview/TriggerLastRunButton.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { connect } from 'react-redux';
+import { impersonateStateToProps } from '@console/internal/reducers/ui';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useAccessReview } from '@console/internal/components/utils';
+import { Button } from 'patternfly-react';
+import { rerunPipeline } from '../../../utils/pipeline-actions';
+import { PipelineModel } from '../../../models';
+
+type TriggerLastRunButtonProps = {
+  pipeline: K8sResourceKind;
+  impersonate?;
+};
+
+const TriggerLastRunButton: React.FC<TriggerLastRunButtonProps> = ({ pipeline, impersonate }) => {
+  const latestRun = _.get(pipeline, ['latestRun'], null);
+  const { label, callback, accessReview } = rerunPipeline(PipelineModel, pipeline, latestRun);
+  const isAllowed = useAccessReview(accessReview, impersonate);
+  return isAllowed ? (
+    <Button variant="secondary" onClick={callback}>
+      {label}
+    </Button>
+  ) : null;
```

Comment:
You almost never need to condition a render between an Element and `null` in React. Falsey values are not rendered. Recommend just simplifying this to:

```
  return (
    isAllowed && (
      <Button variant="secondary" onClick={callback}>
        {label}
      </Button>
    )
  );
```

Alternatively you can just return `null` if not allowed and then after that return the `Button`. Both are fine with me.

Reply:
fixed in https://github.com/openshift/console/pull/3056/commits/8e27cbfc8c376c202bd2471c788bf0401b8dcb9d